lamindb/_curate.py

```diff
@@ -737,7 +748,22 @@ def save_artifact(self, description: str | None = None, **kwargs) -> Artifact:
 
 
 class Curate:
-    """Annotation flow."""
+    """Curation flow.
+
+    Data annotation entails accurately labeling biological datasets with relevant metadata and functional information
+    to facilitate data integration, interpretation and analysis.
+
+    The curation flow has several steps:
+    1. Create an :class:`DataFrameCurator` (:meth:`~lamindb.Curate.from_df`), :class:`AnnDataCurator` (:meth:`~lamindb.Curate.from_anndata`),
```

Comment:
This line is intimidating! What about just listing the `from_` constructors and let users explore return type annotations themselves?

Reply:
Yeah I was 50/50 here while writing this. It will look a tiny bit less scary because the `from_` constructors will not expand but I'll consider removing the types.